cli/cli.go

```diff
@@ -315,45 +316,19 @@ func Execute(version string) {
 	}
 }
 
-//go:embed completions/asdf.bash
-var bashCompletions string
-
-//go:embed completions/asdf.zsh
-var zshCompletions string
-
-//go:embed completions/asdf.fish
-var fishCompletions string
-
-//go:embed completions/asdf.nu
-var nuCompletions string
-
-//go:embed completions/asdf.elv
-var elvishCompletions string
-
 func completionCommand(l *log.Logger, shell string) error {
-	switch shell {
-	case "bash":
-		fmt.Print(bashCompletions)
-		return nil
-	case "zsh":
-		fmt.Print(zshCompletions)
-		return nil
-	case "fish":
-		fmt.Print(fishCompletions)
-		return nil
-	case "nushell":
-		fmt.Print(nuCompletions)
-		return nil
-	case "elvish":
-		fmt.Print(elvishCompletions)
-		return nil
-	default:
-		fmtString := `No completions available for shell with name %s
-Completions are available for: bash, zsh, fish, nushell, elvish`
-		msg := fmt.Sprintf(fmtString, shell)
-		l.Print(msg)
-		return errors.New(msg)
+	file, ok := completions.Get(shell)
+	if !ok {
+		err := fmt.Errorf(`No completions available for shell with name %q
+Completions are available for: %v`, shell, strings.Join(completions.Names(), ", "))
```

Comment:
To be honest, I'm not thrilled with the error message as is. Go error messages usually don't start with with an uppercase letter and are usually fairly short so that they can be chained as `error doing thing: this broke: more details` nicely. I'm not sure why this being turned into an error, though, because I'm not sure what returning it does. Does it simply cause the CLI package to exit with a `1` instead of a `0` and nothing else? If so, I have a cleaner alternative.

Reply:
> Does it simply cause the CLI package to exit with a 1 instead of a 0 and nothing else?

No, it also logs that error message to STDERR via the logger. That STDERR message should bubble up to the user wherever `asdf completion <shell>` is invoked from.